app/controllers/application_controller.rb

```diff
@@ -3,6 +3,8 @@
 class ApplicationController < ActionController::Base
   # Returns the current signed in User (if any)
   def current_user
+    # Overwrites the session cookie if an extended_session cookie exists
+    session[:user_id] = cookies.encrypted[:_extended_session]['user_id'] if cookies.encrypted[:_extended_session].present?
```

Comment:
You can do this only once when the session is empty and the `extended_session` isn't.